src/charm.py

```diff
@@ -332,6 +335,11 @@ def _set_status(self, key: Status) -> None:
         getattr(logger, log_level.lower())(status.message)
         self.unit.status = status
 
+    @property
+    def version(self) -> str:
+        """Reports the current workload (Kafka) version."""
+        return self.workload.get_version()
+
```

Comment:
**nitpick:** Personally I don't think we need this property. I think we can just do this wherever it's needed:
```python
self.unit.set_version(self.workload.get_version())
```